pkgs/applications/science/astronomy/kstars/default.nix

```diff
@@ -9,20 +9,20 @@
 
   eigen, zlib,
 
-  cfitsio, indilib, xplanet, libnova, libraw, gsl, wcslib
+  cfitsio, indilib, xplanet, libnova, libraw, gsl, wcslib, stellarsolver
 }:
 
 mkDerivation rec {
   pname = "kstars";
-  version = "3.4.3";
+  version = "3.5.0";
 
   src = fetchurl {
-    url = "https://mirrors.mit.edu/kde/stable/kstars/kstars-${version}.tar.xz";
-    sha256 = "0j5yxg6ay6sic194skz6vjzg6yvrpb3gvypvs0frjrcjbsl1j4f8";
+    url = "mirror://kde/stable/kstars/kstars-${version}.tar.xz";
+    sha256 = "0fpkm75abn0hhdhfyvpfl6n0fr7gvw63xhb4hvwdrglhkf2nxam1";
   };
 
   patches = [
-    ./indi-fix.patch
+    ./fs-fixes.patch
```

Comment:
This patch should be explained:

- Who wrote the patch?
- Should we expect the next release to not need the patch?
- Is upstream aware of the changes required here for NixOS?

If upstream was submitted the patch, you should use `fetchpatch`.

Reply:
I added a comment to the file in the most recent push explaining its purpose. To answer your questions about its nature/origins:

- I wrote the patch specifically for nix based installations
- Future releases will likely need the patch (though it may need to be reorganized if ksutils.cpp changes)
- I spoke with upstream in their matrix channel about the problem, and they are open to upstreaming a more general solution if me or someone else comes along to fix it.

The nature of the problem is that kstars runs other applications (like indi server for telescope control) as subprocesses, with their locations defined in ksutils.cpp. As it stands now, it uses a single $prefix/bin directory to find all of these binaries, whereas Nix uses different nix store paths for each of these dependencies. The patch lets the nix derivation pass individual nix store paths to cmake like "-DINDI_NIX_ROOT=${indilib}" for each of these programs. 

An alternative route I didnt go would be to have an optional FHS wrapper for kstars that throws them all in the same prefix, but i feel that the patch is a more simple solution for the time being. I'm also trying to work out a better solution upstream, at which point the patch can simply be removed.

Reply:
> I'm also trying to work out a better solution upstream, at which point the patch can simply be removed.

Fair enough.

> An alternative route I didnt go with would be to have an optional buildEnv wrapper for kstars that throws them all in the same prefix, but i feel that the patch is a more simple solution for the time being.

I agree.